ilmercatinodellibro/ilmercatinodellibro
# fix(client+server): fix issues #46, #50, #51, #52, #47 + other bugfixes

## What it does
This PR fixes the following issues:
- #46 
- #50 
- #51 
- #52 
- #47

In addition, it also addresses the following problems:
- Admins can now update the `sellRate` and `purchaseRate` settings at any time, as it is simply entrusted to them not to change them during the Mercatino's activity period.
- A delegate's full name is now mandatory for updating users who are less than 18 years old.
- It is now possible to add multiple copies of the same book to the customer book withdrawals list.
- The sale and purchase prices are now calculated and displayed correctly everywhere.

## How to test it
Check for each issue to be fixed properly; especially important to check the prices display and calculation mechanism

### Additional information
There currently are no user registration arguments that permit the implementation of the constraint for the delegate requirement for minor users. Once the new registration mechanism is handled, the relative check must be added as well. A `TODO` was added in `packages/server/src/modules/user/user.resolver.ts` for this.
Cc @yusufkandemir @SirAuron

Regarding #46: a language switcher has been added on both `/select-locations` and the header of the unlogged UI; this component has not yet been designed so it may be subject to changes in the future.

For completeness's sake: I noticed while working on these tasks that there is also some update logic I missed in #39 about updating the payoff dialog data once the settlement of a user has been done.
